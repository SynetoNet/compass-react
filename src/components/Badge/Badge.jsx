import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import BootstrapBadge from "react-bootstrap/Badge";

import "./Badge.scss";

const Badge = ({ color, variant, width, block, ...props }) => {
  const classes = classNames({
    outline: variant === "outline",
    strong: variant === "strong",
    ["width-" + width]: width,
    block
  });

  return <BootstrapBadge variant={color} className={classes} {...props} pill />;
};

Badge.propTypes = {
  color: PropTypes.oneOf(["success", "danger", "warning", "info", "secondary"]),
  variant: PropTypes.oneOf(["default", "strong", "outline"]),
  width: PropTypes.oneOf([32, 64, 96, 128, "auto"]),
  /** Makes badge stretch to the size of it's parent */
  block: PropTypes.bool
};

Badge.defaultProps = {
  color: "secondary",
  variant: "default",
  width: "auto",
  block: false
};

export default Badge;
