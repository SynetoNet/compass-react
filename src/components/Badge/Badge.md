### **Description**

---

#### **Usage**

Used to communicates various objects states or exceptional situations.

#### **Anatomy**

A badge is composed of:<br/>
Label (regular text)<br/>
Background (color fill)<br/>

### **Color variations**

---

```jsx
<h5>Strong Background</h5>
<Badge color="success">Green</Badge>
<Badge color="danger">Red</Badge>
<Badge color="warning">Yellow</Badge>
<Badge color="primary">Blue</Badge>
<Badge color="secondary">Neutral</Badge>
<br />
<br />
<h5>Light Background</h5>
<Badge color="success" lightBackground>Green</Badge>
<Badge color="danger" lightBackground>Red</Badge>
<Badge color="warning" lightBackground>Yellow</Badge>
<Badge color="primary" lightBackground>Blue</Badge>
<Badge color="secondary" lightBackground>Neutral</Badge>
<br />
<br />
<h5>Light Border</h5>
<Badge color="success" lightBorder>Green</Badge>
<Badge color="danger" lightBorder>Red</Badge>
<Badge color="warning" lightBorder>Yellow</Badge>
<Badge color="primary" lightBorder>Blue</Badge>
<Badge color="secondary" lightBorder>Neutral</Badge>
```

### **Sizes**

---

```jsx
<br />
<br />
<h5>Modular Size Badges</h5>
<Badge color="success" modularWidth widthSize={32}>32px</Badge>
<br />
<Badge color="success" modularWidth widthSize={64}>64px</Badge>
<br />
<Badge color="success" modularWidth widthSize={96}>96px</Badge>
<br />
<Badge color="success" modularWidth widthSize={128}>128px</Badge>
<br />
<br />
<h5>Elastic Size Badges</h5>
<Badge color="success" elasticWidth>Short</Badge>
<br />
<Badge color="warning" elasticWidth>Medium Size</Badge>
<br />
<Badge color="danger" elasticWidth>Longer ass Badge</Badge>
<br />
<br />
```
