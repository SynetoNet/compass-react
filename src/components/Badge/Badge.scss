@import "../../base/colors";
@import "../../base/variables";

.badge {
  height: 20px;
  width: auto;
  display: inline-block;
  padding: 0 12px;
  font-size: 12px;
  font-weight: normal;
  line-height: 20px;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;

  &.outline {
    // it contains borders, so content is only 22px height
    line-height: 18px;
  }
  &.block {
    display: block;
  }
}

.badge-success {
  background-color: $semantic-confirm-light-85;
  color: $semantic-confirm-dark-30;
  &.strong {
    background-color: $semantic-confirm;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $semantic-confirm-light-60;
  }
}
.badge-danger {
  background-color: $semantic-danger-light-85;
  color: $semantic-danger-dark-30;
  &.strong {
    background-color: $semantic-danger;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $semantic-danger-light-60;
  }
}
.badge-info {
  background-color: $semantic-info-light-85;
  color: $semantic-info-dark-30;
  &.strong {
    background-color: $semantic-info;
    color: $neutral-0;
  }
  &.outline {
    border: 1px solid $semantic-info-light-60;
    background-color: $neutral-0;
  }
}
.badge-warning {
  background-color: $semantic-warn-light-85;
  color: $semantic-warn-dark-30;
  &.strong {
    background-color: $semantic-warn;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $semantic-warn-light-60;
  }
}
.badge-secondary {
  background-color: $neutral-12;
  color: $neutral-90;
  &.strong {
    background-color: $neutral-60;
    color: $neutral-0;
  }
  &.outline {
    background-color: $neutral-0;
    border: 1px solid $neutral-12;
    color: $neutral-40;
  }
}

.width-32 {
  width: $spacer * 8;
  padding: 0 0;
}
.width-64 {
  width: $spacer * 16;
  padding: 0 0;
}
.width-96 {
  width: $spacer * 24;
  padding: 0 0;
}
.width-128 {
  width: $spacer * 32;
  padding: 0 0;
}
.width-160 {
  width: $spacer * 40;
  padding: 0 0;
}
.width-auto {
  width: auto;
}
