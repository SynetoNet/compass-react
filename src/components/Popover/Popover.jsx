import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import BSPopover from "react-bootstrap/Popover";
import "./Popover.scss";

class Popover extends React.Component {
  render() {
    const { content, placement, trigger, title, ...props } = this.props;
    return (
      <OverlayTrigger
        placement={placement}
        show={"false"}
        {...getTriggerProp(trigger)}
        overlay={
          <BSPopover title={title} {...props}>
            {content}
          </BSPopover>
        }
      >
        {this.props.children}
      </OverlayTrigger>
    );
  }
}

function getTriggerProp(trigger) {
  // force close on click outside
  if (trigger === "click") {
    return { trigger, rootClose: true };
  }
  return { trigger };
}

Popover.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.node.isRequired,
  placement: PropTypes.oneOf(["top", "bottom", "left", "right", "auto"]),
  trigger: PropTypes.oneOf(["focus", "click", "hover"])
};

Popover.defaultProps = {
  placement: "auto",
  trigger: "hover"
};

export default Popover;
