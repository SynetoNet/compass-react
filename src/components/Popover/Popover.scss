.popover-container {
  display: inline-block;
  // width: 500px;
}
.popover {
  // max-width: auto;
}