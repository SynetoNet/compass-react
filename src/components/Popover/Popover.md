#### **Default**

```jsx
import Badge from "../Badge/Badge";

<Popover title="Some additional info" content="Some content">
  <Badge variant="strong">Hover over me!</Badge>
</Popover>;
```

#### **Placement**

```jsx
import Badge from "../Badge/Badge";
<>
  <Popover
    placement="top"
    title="Some additional info"
    trigger="hover"
    content="Some content"
  >
    <Badge variant="strong">Popover on top</Badge>
  </Popover>
  <Popover
    placement="bottom"
    title="Some additional info"
    trigger="hover"
    content="Some content"
  >
    <Badge variant="strong">Popover on bottom</Badge>
  </Popover>
  <Popover
    placement="right"
    title="Some additional info"
    trigger="hover"
    content="Some content"
  >
    <Badge variant="strong">Popover on the right</Badge>
  </Popover>
  <Popover
    placement="left"
    title="Some additional info"
    trigger="hover"
    content="Some content"
  >
    <Badge variant="strong">Popover on the left</Badge>
  </Popover>
</>;
```

#### **Trigger**

```jsx
import Button from "../Button/Button";
<>
  <div>
    Click -
    <Popover
      placement="right"
      title="Some additional info"
      trigger="click"
      content="Some content"
    >
      <Button variant="primary">Click me!</Button>
    </Popover>
  </div>
  <div>
    Hover -
    <Popover
      placement="right"
      title="Some additional info"
      trigger="hover"
      content="Some content"
    >
      <Button variant="primary">Hover over me!</Button>
    </Popover>
  </div>
  <div>
    Focus -
    <Popover
      placement="right"
      title="Some additional info"
      trigger="focus"
      content="Some content"
    >
      <Button variant="primary">Focus me!</Button>
    </Popover>
  </div>
</>;
```

#### **Extra content**

```jsx
import Button from "../Button/Button";

<Popover
  placement="right"
  title="Popover title"
  content={
    <>
      <div>Here's an example of text and another component</div>
      <br />
      <Button variant="primary">Click me!</Button>
    </>
  }
  trigger="click"
>
  <Button variant="primary">Click me!</Button>
</Popover>;
```
