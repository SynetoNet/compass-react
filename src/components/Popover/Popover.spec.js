import React from "react";
import Popover from "./Popover";
import Badge from "../Badge/Badge.jsx";
import renderer from "react-test-renderer";

import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Default Popover", () => {
  const defaultExample = (
    <Popover title="Default popover">
      <Badge variant="strong">Hover over me!</Badge>
    </Popover>
  );

  test("popover snapshop", () => {
    const component = shallow(defaultExample);
    expect(component).toMatchSnapshot();
  });
});
