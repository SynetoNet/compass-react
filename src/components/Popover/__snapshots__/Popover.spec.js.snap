// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Popover popover snapshop 1`] = `
<div
  className="popover-container"
>
  <OverlayTrigger
    defaultShow={false}
    overlay={
      <ForwardRef
        content="content"
        placement="right"
        style={null}
      >
        <ForwardRef>
          title
        </ForwardRef>
        <ForwardRef />
      </ForwardRef>
    }
    placement="auto"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <span>
      trigger
    </span>
  </OverlayTrigger>
</div>
`;
